addons/website_slides/models/slide_channel.py

```diff
@@ -46,24 +48,46 @@ class ChannelUsersRelation(models.Model):
         )
     ]
 
+    def _compute_next_slide_id(self):
+        read_group_res = self.env['slide.slide.partner'].sudo()._read_group(
+            ['&', '&', ('channel_id', 'in', self.mapped('channel_id').ids),
+             ('partner_id', 'in', self.mapped('partner_id').ids),
+             ('completed', '=', True),
+             ('slide_id.is_published', '=', True),
+             ('slide_id.active', '=', True)],
+            ['slide_id:array_agg'], ['channel_id', 'partner_id'], lazy=False)
```

Comment:
I don't like `slide_id:array_agg` :/
Basically you will retrieve all ids but you need only the first one

I think using `slide_id:max` directly won't be correct as well (create_date should have priority over id) so I think you need to do a SQL query